dev/com.ibm.ws.jmx.connector.local/src/com/ibm/ws/jmx/connector/local/LocalConnectorActivator.java

```diff
@@ -62,26 +62,50 @@ public String run() {
                             System.setProperty(RMI_SERVER_HOSTNAME_PROPERTY, LOOPBACK_ADDRESS);
                         }
 
+                        String localConnectorAddress = null;
                         // Start the JMX agent and retrieve the local connector address.
-                        // TODO: Find a proper way to get the JMX agent's local connector address
-                        //       for now we need to depend on JDK internal APIs
-                        ClassLoader systemClassLoader = ClassLoader.getSystemClassLoader();
-                        Class<?> clazz1 = JavaInfo.majorVersion() < 9 
-							? Class.forName("sun.management.Agent", true, systemClassLoader) 
-							: Class.forName("jdk.internal.agent.Agent", true, systemClassLoader);
-                        Method m1 = clazz1.getMethod("agentmain", String.class);
-                        m1.invoke(null, (Object) null);
-                        String localConnectorAddress = System.getProperty(LOCAL_CONNECTOR_ADDRESS_PROPERTY);
-                        if (localConnectorAddress == null) {
-                            Class<?> clazz2 = JavaInfo.majorVersion() < 9 
-								? Class.forName("sun.misc.VMSupport", true, systemClassLoader) 
-								: Class.forName("jdk.internal.vm.VMSupport", true, systemClassLoader);
-                            Method m2 = clazz2.getMethod("getAgentProperties");
-                            Properties props = (Properties) m2.invoke(null);
-                            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                                Tr.debug(tc, "Attempting to retrieve the connector address from agent properties.");
+                        if (JavaInfo.majorVersion() < 9 ||
+                            (JavaInfo.majorVersion() >= 9 && !Boolean.getBoolean("jdk.attach.allowAttachSelf") ||
+                             System.getSecurityManager() != null)) {
```

Comment:
@aguibert we think the problem is here ... if `System.getSecurityManager() != null` then it should use `jdk.internal.agent.Agent`??

Reply:
Kinda. We think that if `JavaInfo.majorVersion() >= 9` and this whole condition is also true, then on line 79 we should be using `jdk.internal.agent.Agent` and on line 84 we should be using `jdk.internal.vm.VMSupport`, as the old code did.

Reply:
yep, @Azquelt is spot on -- my changes were preserving existing behavior for the Java 7/8 code path but not the Java 11+ code path